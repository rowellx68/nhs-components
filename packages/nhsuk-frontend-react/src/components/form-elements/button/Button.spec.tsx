import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  Primary as PrimaryStory,
  PrimaryButtonAsLink as PrimaryButtonAsLinkStory,
} from './Button.stories';

const Button = composeStory(PrimaryStory, meta);
const ButtonAsLink = composeStory(PrimaryButtonAsLinkStory, meta);

it('should render the Button component as a button', () => {
  const { container } = render(<Button>Submit</Button>);

  expect(container.querySelector('button')).toBeInTheDocument();
  expect(container).toHaveTextContent('Submit');
});

it.each(['secondary', 'reverse'] as const)(
  'should render the Button component with the %s variant',
  (variant) => {
    const { container } = render(<Button variant={variant}>Action</Button>);

    expect(
      container.querySelector('.nhsuk-button--' + variant),
    ).toBeInTheDocument();
  },
);

it('should render the Button component with the disabled attribute', () => {
  const { container } = render(<Button disabled>Disabled</Button>);

  expect(container.querySelector('button')).toHaveAttribute('disabled');
});

it('should render the Button component as a link', () => {
  const { container } = render(
    <ButtonAsLink as="a" href="https://www.nhs.uk">
      Link
    </ButtonAsLink>,
  );

  expect(container.querySelector('a')).toBeInTheDocument();
});

it.each(['secondary', 'reverse'] as const)(
  'should render the Button component as a link variant %s',
  (variant) => {
    const { container } = render(
      <ButtonAsLink as="a" href="https://www.nhs.uk" variant={variant}>
        Link
      </ButtonAsLink>,
    );

    expect(container.querySelector('a')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
    expect(
      container.querySelector('.nhsuk-button--' + variant),
    ).toBeInTheDocument();
  },
);
