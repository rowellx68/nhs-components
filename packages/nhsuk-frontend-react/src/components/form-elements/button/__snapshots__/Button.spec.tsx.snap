// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the Button component as a link variant reverse 1`] = `
<div>
  <a
    class="nhsuk-button nhsuk-button--reverse"
    data-module="nhsuk-button"
    draggable="false"
    href="https://www.nhs.uk"
    role="button"
  >
    Link
  </a>
</div>
`;

exports[`should render the Button component as a link variant secondary 1`] = `
<div>
  <a
    class="nhsuk-button nhsuk-button--secondary"
    data-module="nhsuk-button"
    draggable="false"
    href="https://www.nhs.uk"
    role="button"
  >
    Link
  </a>
</div>
`;

exports[`should render the Button component as a link variant warning 1`] = `
<div>
  <a
    class="nhsuk-button nhsuk-button--warning"
    data-module="nhsuk-button"
    draggable="false"
    href="https://www.nhs.uk"
    role="button"
  >
    Link
  </a>
</div>
`;
